interface/src/app/volcano-plot/volcano-plot.js

```diff
@@ -1,35 +1,23 @@
 /*
  * The volcano-plot module provides a directive and controller for embedding
- * volcano plots in other views. Requires samples to be grouped in the
- * adage.analyze.sampleBin in order to function properly -- if no samples are
- * chosen or placed into groups, an error will be displayed.
+ * volcano plots in other views. Requires properly prepared data to be specified
+ * via plotData.
  */
 angular.module('adage.volcano-plot', [
   'ngVega',
   'greenelab.stats',
-  'adage.analyze.sampleBin',
   'adage.volcano-plot-vgspec'
 ])
 
-.controller('VolcanoPlotCtrl', ['$scope', 'VolcanoPlotSpec', 'SampleBin',
-  function VolcanoPlotCtrl($scope, VolcanoPlotSpec, SampleBin) {
-    // retrieve the nodeInfo map (for node names) via the SampleBin service
-    // $scope.nodeInfo = SampleBin.getNodeInfo(); // TBD
-    $scope.nodeInfo = [
-      {'id': 1, 'name': 'Node1pos'},
-      {'id': 2, 'name': 'Node2pos'},
-      {'id': 518, 'name': 'test name'}
-    ];
-
+.controller('VolcanoPlotCtrl', ['$scope', 'VolcanoPlotSpec',
+  function VolcanoPlotCtrl($scope, VolcanoPlotSpec) {
     // define controller instance variables that link with vega
-    this.data = {
-      source: $scope.plotData
-      // FIXME: this is not working as it should to bring in node names
-      // nodes: $scope.nodeInfo
-      // FIXME: can't get selectedNodes out this way... need to use view api?
-      // selectedNodes: $scope.selectedNodes
-    };
     this.spec = VolcanoPlotSpec;
+    this.data = $scope.plotData;
```

Comment:
Okay, so I thought I knew what I was doing here, then I got totally lost in Angular documentation trying to unravel a semi-related issue with the volcano plot selections, and now I'm back here because I think I know what I'm doing here again.

You are totally right, @rzelayafavila, for pointing out that this is a kinda funky thing to do: use *both* the `controllerAs` syntax *and* define `$scope` to access an isolate scope. What results is that ugly duckling line of code (`line 16`) that we're talking about here. The reason for doing each one is that these steps take us a little closer to more of a `.component` structure. With `controllerAs`, we're defining a name space for the controller that's easily accessed from the template in a clear way. With the isolate scope, we are preventing the component from using or modifying anything from the parent scope inadvertently.

What is not quite right, however, is that I could *not* access the value directly from `this.plotData`, as you suggest. It's not there. I *could* access `this.data` by referring to it as `$scope.vp.data`, because what `controllerAs` does, in effect, is to insert `$scope.vp = this;` into the first line of the controller. (See [this article](http://codetunnel.io/angularjs-controller-as-or-scope/) that @dongbohu posted to Slack a while back.) But the opposite is not true. To access the contents of the isolate scope I need to have access to `$scope`.

But wait... there's more. If we *wanted* to have the contents of the isolate scope bound to the controller, then we could include the property `bindToController: true` in our [directive definition object](https://docs.angularjs.org/api/ng/service/$compile#directive-definition-object), as described in [this article](https://toddmotto.com/no-scope-soup-bind-to-controller-angularjs/) and [this other article](https://www.martin-brennan.com/angular-directive-isolate-scope-explained/).

Exhausted yet? I am. This whole mess is partly why `.component` was invented, I believe. I will solve this by migrating from a `.directive` to `.component`, something that might happen as soon as the next pull request. But not this pull request, because the code is working. Does that work for you, @rzelayafavila?